### ISSUE TITLE
Can't connect 'containerView' and 'buttonBarView' outlets

### ISSUE THREAD

**Issue by joaohkfaria** [xmartlabs/XLPagerTabStrip#141]
Hi! I'm trying to connect the outlets for containerView and buttonBarView, but this outlets doesn't exists on my class, only on it's parent (BaseButtonBarPagerTabStripViewController). But I can't connect, it just doesn't work.

How can I proceed?

<img width="616" alt="screen shot 2016-03-11 at 2 53 36 pm" src="https://cloud.githubusercontent.com/assets/1427347/13710591/17d4f326-e799-11e5-8410-97745f7c35e6.png">

**Comment by mtnbarreto** [comment]
This is because `TabsViewController` is a generic class and IB does not work with generics.

As a workaround you can change the base class of `TabsViewController` to `ButtonBarPagerTabStripViewController` then connect the `ButtonBarPagerTabStripViewController` outlets then put back your parent generic class (BaseButtonBarPagerTabStripViewController<TabIconCellView>). 

You will also have to invoke `let _ = TabsViewController(nibName: nil, bundle: nil)` from `func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool` in in order to load the your generic class into objective-c runtime.

Doing that I made Youtube example works in the example app.

Hope it helps you.

**Comment by joaohkfaria** [comment]
Ok, I did what you've said and using `ButtonBarPagerTabStripViewController` I could connect `containerView` and `buttonBarView` (and it's working). Then, I've changed back my `TabsViewController` to the YouTube example (and followed your advice adding that line on `AppDelegate`).

But it seems that is rejecting the containerView outlet. Any ideas?

Error Message:
> *** Terminating app due to uncaught exception 'NSUnknownKeyException', reason: '[<UIViewController 0x7c286420> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key containerView.'

**Comment by mtnbarreto** [comment]
Have you deleted 

`buttonbarView and `containerView` properties/outlets from your TabsViewController code. You should connect the base class outlets and not re-declare these properties/outlets in your own class.

**Comment by joaohkfaria** [comment]
Yes, I'm doing that. I've connected them from the base class (`containerView` from `PageTabStripViewController` and `buttonBarView` from `ButtonBarPagerTabStripViewController`)

**Comment by mtnbarreto** [comment]
Could you upload a base project so i can take a look..? 
I have done this twice and it worked in both. 

Is the storyboard view controller `Class`  properly set up ? I'm asking because when you extend from the generic base controller IB does not suggest it when typing.

**Comment by joaohkfaria** [comment]
I can't send you my project, but I can send you the two classes that I'm working.

Here it is:
- TabsViewController
http://paste.ofcode.org/aXcxvAJSdLJUTUyGuiUJka
- EventListViewController (Indicator)
http://paste.ofcode.org/NJNHbimdz4VEYsCD2BGFti

**Comment by mtnbarreto** [comment]
`let _ = TabsViewController(nibName: nil, bundle: nil)` must be executed before instantiating the storyboard view controller. 

Is `TabsViewController` the root view controller?

**Comment by mtnbarreto** [comment]
I just tried setting up YoutubeExampleViewController as root view controller and it does not work. 

Could you try adding another view controller as a root view controller and navigating to your tab bar view controller in some way ?

**Comment by joaohkfaria** [comment]
Yes, it is!

![screen shot 2016-03-11 at 6 42 11 pm](https://cloud.githubusercontent.com/assets/1427347/13716305/05f07066-e7b9-11e5-9d40-59099342aa87.png)

**Comment by mtnbarreto** [comment]
As a workaround in order to make it work as initial view controller i had to...

* remove initial view controller arrow from storyboard. 
* change my app delegate code as shown below:

```swift
    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
        ..
        let _ = YoutubeExampleViewController(nibName: nil, bundle: nil)
        
        let sb = UIStoryboard(name: "Storyboard", bundle: nil)
        let tabBarViewController = sb.instantiateViewControllerWithIdentifier("YoutubeControllerId")
        window = UIWindow(frame: UIScreen.mainScreen().bounds)
        window?.rootViewController = tabBarViewController
        window?.makeKeyAndVisible()
        
        return true
    }
```

**Comment by joaohkfaria** [comment]
Ok! Now it's working like a charm! Thank you very much @mtnbarreto :)

**Comment by narlei** [comment]
This not work in app delegate in Objective-C. Can you help me?

**Comment by tin612** [comment]
thank you so much @mtnbarreto . Have been stuck in that thing for 2 hours

**Comment by keason** [comment]
How Can I fix the problem if the Youtube Class is child class of TabbarViewController?

**Comment by ZeranLatte** [comment]
@keason I am trying to do this as well because I need customized button cells, any updates? I am able to connect the buttonBarView and containView from sb to my viewController: BaseButtonBarPagerTabStripViewController<CustomizedCell>; but having the KVC error when I tab this tabitem from UITabbar.

**Comment by ZeranLatte** [comment]
@keason I have solved this by applying the workaround from mtnbarreto. Another issue I had is solved after checking #355
